app/views/play/VoyagerUnitMap.vue

```diff
@@ -0,0 +1,211 @@
+        const classroom = this.classroom = await api.classrooms.get({classroomID: classroomId})
+        const classroomCourseLevels =  _.find(classroom.courses, {_id: courseId}).levels
+
+        // get levels data for the levels in the classroom snapshot
+        const classroomCourseLevelsData = await api.classrooms.getCourseLevels({classroomID: classroomId, courseID: courseId})
+
+        for (let level of classroomCourseLevels) {
+          this.classroomLevelMap[level.original] = level
+        }
+        
+        let courseLevelsData = {}
+        for (let level of classroomCourseLevelsData) {
+          let original = level.original
+          if (existingCampaignLevels[original]) {
+            courseLevelsData[original] = existingCampaignLevels[original]
+          }
+          else {
+            // a level which has been removed from the campaign but is saved in the course
+            courseLevelsData[original] = level
+          }
+          // carry over position, nextLevels, first property stored in classroom course, if there are any
+          if (this.classroomLevelMap[original].position)
+            courseLevelsData[original].position = this.classroomLevelMap[original].position
+          if (this.classroomLevelMap[original].nextLevels)
+            courseLevelsData[original].nextLevels = this.classroomLevelMap[original].nextLevels
+          if (this.classroomLevelMap[original].first)
+            courseLevelsData[original].first = this.classroomLevelMap[original].first
+        }
+        // TODO: what if level was removed from campaign, and classroom level map doesnt contain positions, nextLevels, and first? Is it a valid scenario?
+        // If yes, then use the logic from CampaignView for setting the missing positions using the prev and next level positions
+        // Not sure how to set the .first and .nextLevels in that case.
+
+        return courseLevelsData
+      }
+      catch (err) {
+        return Promise.reject(err)
+      }
+    },
+    createLevelStatusMap() {
+      // Remove the level sessions for the levels played in another language - for the classroom version of unit map
+      // TODO: What if played in same language in another classroom
+      // TODO: What should be done for home version (i.e. played a level in another campaign)
+      if(this.classroomLevelMap && this.classroom) {
+        for (let session of this.levelSessions) {
+          const classroomLevel = this.classroomLevelMap[session.level.original]
+          if (!classroomLevel)
+            continue
+          const expectedLanguage = classroomLevel.primerLanguage || this.classroom.aceConfig.language
+          if (session.codeLanguage != expectedLanguage) {
+            this.levelSessions.splice(this.levelSessions.indexOf(session),1)
+          } 
+        }
+      }
+      this.levelStatusMap = utils.getLevelStatusMap(this.levelSessions)
+    },
+    determineNextLevel() { // set .next and .locked for this.levels      
```

Comment:
In a second reading of this PR I can now see the differences between this `determineNextLevel` method and the `getNextLevelVoyager` method. I am also starting to see and experience the challenge of consolidating these various situations.

The `determineNextLevel` method has the following responsibilities:
 - Unlock levels for the user to play.
 - Also sets `.next` as a side effect.

Will the `.next` property *always* be in sync with what the `getNextLevelVoyager` returns?
An alternative is having a method to return levels to unlock. And then having a method to return next levels.
By decoupling these methods we can test them as separate concepts.

Finally in order to change state, updating the campaign view. We could layer these methods with different intentions and mutate the state based on their results.

The problem that I am trying to express is:
 - How can we ensure the various functions that are overlapping in functionality **always** agree?
 - Is there a chance that someone in the future updates this method and it is thus out of sync with `getNextLevelVoyager`?

Reply:
These are great questions.
I think that finding the next level in `findNextLevelsBySession` and then setting `.next` based on that here in `determineNextLevel` doesnt look like could get out of sync, since we set `.next` on whatever is returned from `findNextLevelsBySession`
However, I agree that setting `.locked` and setting `.next` logic can be decoupled a little more.
How about splitting `determineNextLevel` to something like this:
For unlocking levels:
```
for (let level in this.levels) {
          if (this.levelStatusMap[level] == 'started' || this.levelStatusMap[level] == 'complete' || this.levels[level].first || nextLevelOriginals.includes(level))
            this.levels[level].locked = false
          else
            this.levels[level].locked = true
        }
```
And for next levels:
```
for (let level in this.levels) {
          // there would only be one next level as per voyager v1 as of now
          if (nextLevelOriginals.includes(level)){
            this.levels[level].next = true
          }
        }
```
Not sure if I understand your questions right, and if this would address them?

Reply:
This makes the two intentions more readable and I like it for that reason.
Instead of two loops, they could be two separate map operations.
Your example is clearer as it keeps functionality separate with minimal runtime cost

Reply:
This makes the two intentions more readable and I like it for that reason.
Instead of two loops, they could be two separate map operations.
Your example is clearer as it keeps functionality separate with minimal runtime cost